### PR TITLE
fix(engine): don't handle pull request with more than 500 check-runs

### PR DESCRIPTION
**Pull request by sileht**
Fixes MRGFY-1076

### CODE REVIEW

File: mergify_engine/clients/github.py [hunk 1]
```diff
@@ -431,6 +431,7 @@ async def items(
         oauth_token: typing.Optional[github_types.GitHubOAuthToken] = None,
         list_items: typing.Optional[str] = None,
         params: typing.Optional[typing.Dict[str, str]] = None,
+        page_limit: int = 100,
```
Comment (jd) [inline mergify_engine/clients/github.py:434]: Wondering if we should decrease that in another PR.
I'm having a hard time imagining what will work with 100 pages, whatever the request is.

File: mergify_engine/check_api.py [hunk 2]
```diff
@@ -162,9 +164,15 @@ async def get_checks_for_ref(
...
                     list_items="check_runs",
                     params=params,
+                    page_limit=5,
                 ),
             )
         ]
+    except github.TooManyPages as e:
+        raise exceptions.UnprocessablePullRequest(
+            f"The pull request reports more than {(e.last_page - 1) * 100} check-runs, "
+            "reaching the limit of 500 check-runs."
```
Comment (jd) [inline mergify_engine/check_api.py:174]:
```suggestion
            f"The pull request reports more than {(e.last_page - 1) * 100} check runs, "
            "reaching the limit of {page_limit * 100} check runs."
```

**Review by jd — DISMISSED** [review]

File: mergify_engine/clients/github.py (hunk 1 above)
Comment (sileht) [reply mergify_engine/clients/github.py:436]: Agreed!

Comment (sileht) [reply mergify_engine/clients/github.py:436]: I will propose another PR to fix that.

**Review by jd — APPROVED** [review]

**Pull request merged**